ArtOfIllusion/src/artofillusion/preferences/AppearancePreferencesPanelImpl.java

```diff
@@ -0,0 +1,189 @@
+
+import artofillusion.ArtOfIllusion;
+import artofillusion.ui.ThemeManager;
+import artofillusion.ui.ThemeManager.ThemeInfo;
+import artofillusion.ui.Translate;
+
+import javax.swing.*;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Locale;
+import java.util.stream.Collectors;
+
+/**
+ *
+ * @author MaksK
+ */
+
+public class AppearancePreferencesPanelImpl extends javax.swing.JPanel {
+    
+    private static final Comparator<ThemeInfo> tc  = Comparator.comparing(ThemeInfo::getName);
+    private final List<ThemeInfo> themes = ThemeManager.getThemes().stream().filter(info -> info.selectable).sorted(tc).collect(Collectors.toList());
+
+    private final Locale[] languages = Translate.getAvailableLocales();
+    /**
+     * Creates new form AppearancePreferencesPanelImpl
+     */
+    public AppearancePreferencesPanelImpl() {
+        initComponents();
+    }
+
+    private DefaultComboBoxModel<String> getLocalesModel() {
+        DefaultComboBoxModel<String> dcm = new DefaultComboBoxModel<>();
+        Locale cloc =  ArtOfIllusion.getPreferences().getLocale();
+        for(Locale loc: languages) {
+            String localeDisplayName = loc.getDisplayName(loc);
+            dcm.addElement(localeDisplayName);
+            if(cloc.equals(loc)) dcm.setSelectedItem(localeDisplayName);
+        }
+        return  dcm;
+    } 
+    
+    private DefaultComboBoxModel<String> getThemesModel() {
+        DefaultComboBoxModel<String> dcm = new DefaultComboBoxModel<>();
+        String selectedThemeName = ThemeManager.getSelectedTheme().getName();
+        themes.forEach(theme -> {
+            dcm.addElement(theme.getName());
+            if(theme.getName().equals(selectedThemeName)) dcm.setSelectedItem(theme.getName());
+        });
+        return dcm;
+    } 
+      
+    private DefaultComboBoxModel<String> getColorSetModel(ThemeInfo theme) {
+        DefaultComboBoxModel<String> dcm = new DefaultComboBoxModel<>();
+        for(ThemeManager.ColorSet cSet:  theme.getColorSets()) {
+            dcm.addElement(cSet.getName());
+        } 
+        return dcm;
+    } 
+    
+    
+    public Locale getSelectedLocale() {
+        return languages[languageSelector.getSelectedIndex()];
+    }
+
+    public String getSelectedThemeName() {
+        return themeSelector.getSelectedItem().toString();
+    }
+
+    public int getSelectedColorSetIndex() {
+        return colorSetSelector.getSelectedIndex();
+    } 
+    
+    public ThemeInfo getSelectedTheme() {
+        return themes.get(themeSelector.getSelectedIndex());
+    } 
+    /**
+     * This method is called from within the constructor to initialize the form.
+     * WARNING: Do NOT modify this code. The content of this method is always
+     * regenerated by the Form Editor.
+     */
+    @SuppressWarnings("unchecked")
+    // <editor-fold defaultstate="collapsed" desc="Generated Code">//GEN-BEGIN:initComponents
+    private void initComponents() {
```

Comment:
What form editor did you use to generate this code?

Reply:
This is Netbeans GUI editor generated code as well as xml *.form file

Reply:
Please nothing generated by an IDE specific tool. We don't want to lock in to a specific developer toolchain like that...